File: src/main/java/fi/aalto/cs/apluscourses/model/Library.java
package fi.aalto.cs.apluscourses.model;

import org.jetbrains.annotations.NotNull;

import java.util.Collections;
import java.util.List;

public abstract class Library extends Component {

  public Library(@NotNull String name) {
    super(name);
  }

  @NotNull
  @Override
  protected List<String> computeDependencies() {
    return Collections.emptyList();
  }

  @Override
  public boolean isUpdatable() {
    return false;
  }

  @Override
  public boolean hasLocalChanges() {
    return false;
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/model/component/Library.kt
package fi.aalto.cs.apluscourses.model.component

import com.intellij.openapi.project.Project
import com.intellij.openapi.roots.libraries.LibraryTable
import com.intellij.openapi.roots.libraries.LibraryTablesRegistrar
import java.nio.file.Path
import com.intellij.openapi.roots.libraries.Library as IdeaLibrary

abstract class Library(name: String, project: Project) : Component<IdeaLibrary>(name, project) {
    override val platformObject: IdeaLibrary?
        get() = libraryTable(project).getLibraryByName(name)

    override val path: Path
        get() = Path.of(".libs", name)

    override val fullPath: Path
        get() = Path.of(project.basePath!!).resolve(path)

    override fun findDependencies(): Set<String> = emptySet()

    override fun updateStatus() {
        if (status == Status.LOADING) return
        status = if (platformObject != null) {
            Status.LOADED
        } else {
            Status.UNRESOLVED
        }
    }

    companion object {
        fun libraryTable(project: Project): LibraryTable = LibraryTablesRegistrar.getInstance().getLibraryTable(project)
    }
}